### PR TITLE
Use forward-declarations in big editor classes

### PR DESCRIPTION
**Pull request by trollodel**
Follow up of #60684

Reduce the number of includes in some editor classes that are included a lot.

Add forward declarations in touched headers.
Remove some unused includes.

### CODE REVIEW

**Review by KoBeWi — APPROVED** [review]
Nice cleanup again.

I noticed you added some new includes in some `.h` files (editor node, canvas item editor, script text editor etc.). I assume you plan to handle them in follow-up PR(s)?

**Comment by trollodel** [comment]
> I noticed you added some new includes in some .h files (editor node, canvas item editor, script text editor etc.). I assume you plan to handle them in follow-up PR(s)?

Not really. They need non-trivial refactoring (and in some cases you can't e.g., with subclasses), which I don't plan to do. What I'll do in follow-up PRs is to further forward declare stuff.

**Comment by KoBeWi** [comment]
But includes are only needed when you inherit the class or define a method inside header. I checked these files and it doesn't seem to be the case there.

**Comment by trollodel** [comment]
> I checked these files and it doesn't seem to be the case there.

EditorData [doesn't agree with you](https://github.com/trollodel/godot/blob/c5c48dd5549f009909d4d7d690720b593c698cbc/editor/editor_node.h#L724) 😀

> But includes are only needed when you inherit the class or define a method inside header

Another thing that can't be included are enums defined in classes. This is why, e.g., EditorNode [needs `editor_plugin.h`](https://github.com/trollodel/godot/blob/c5c48dd5549f009909d4d7d690720b593c698cbc/editor/editor_node.h#L949).
This is what I gather from a quick search. If there is some ~~obscure~~ C++ syntax for it, then I'm happy to use that and remove the includes.

File: editor/editor_node.h [hunk 1]
```diff
@@ -34,10 +34,11 @@
 #include "core/templates/safe_refcount.h"
 #include "editor/editor_folding.h"
 #include "editor/editor_native_shader_source_visualizer.h"
+#include "editor/editor_plugin.h"
 #include "editor/editor_run.h"
 #include "editor/editor_title_bar.h"
 #include "editor/export/editor_export.h"
-#include "editor/inspector_dock.h"
+#include "editor_data.h"
```
Comment (akien-mga) [inline editor/editor_node.h:41]: Should be `editor/editor_data.h` like other editor includes.

File: editor/plugins/tiles/tile_data_editors.cpp [hunk 2]
```diff
@@ -30,6 +30,9 @@
 
 #include "tile_data_editors.h"
 
+#include "scene/gui/menu_button.h"
+#include "scene/gui/option_button.h"
+#include "scene/gui/separator.h"
 #include "tile_set_editor.h"
```
Comment (akien-mga) [inline editor/plugins/tiles/tile_data_editors.cpp:36]: The intent here was likely to keep the "local" tileset includes first, before core/editor. scene includes should thus go further down (core + editor + scene could be grouped like we usually do).

File: editor/plugins/tiles/tile_map_editor.cpp [hunk 3]
```diff
@@ -30,6 +30,7 @@
 
 #include "tile_map_editor.h"
 
+#include "editor/editor_settings.h"
```
Comment (akien-mga) [inline editor/plugins/tiles/tile_map_editor.cpp:33]: Same issue, this should go with other editor includes.

File: editor/plugins/tiles/tile_data_editors.cpp (hunk 2 above)
Comment (akien-mga) [reply editor/plugins/tiles/tile_data_editors.cpp:36]: Seeing other `tiles` files, it seems @groud's preferred include style is:
```
// class header

// tile-related (local headers)

// editor

// scene

// core
```
That's not exactly the usual include style (https://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html#header-includes) but if it's consistent within `tiles` it makes sense.

File: editor/plugins/version_control_editor_plugin.cpp [hunk 4]
```diff
@@ -39,6 +39,7 @@
 #include "editor/editor_settings.h"
 #include "editor/filesystem_dock.h"
 #include "scene/gui/separator.h"
+#include "script_editor_plugin.h"
```
Comment (akien-mga) [inline editor/plugins/version_control_editor_plugin.cpp:42]: Use absolute paths (`editor/...`).

File: modules/gltf/editor/editor_scene_importer_blend.cpp [hunk 5]
```diff
@@ -29,6 +29,7 @@
 /*************************************************************************/
 
 #include "editor_scene_importer_blend.h"
+#include "scene/gui/line_edit.h"
```
Comment (akien-mga) [inline modules/gltf/editor/editor_scene_importer_blend.cpp:32]:
```suggestion
#include "editor_scene_importer_blend.h"

#include "scene/gui/line_edit.h"
```

File: editor/plugins/texture_layered_editor_plugin.cpp [hunk 6]
```diff
@@ -29,6 +29,7 @@
 /*************************************************************************/
 
 #include "texture_layered_editor_plugin.h"
+#include "scene/gui/label.h"
```
Comment (akien-mga) [inline editor/plugins/texture_layered_editor_plugin.cpp:32]:
```suggestion
#include "texture_layered_editor_plugin.h"

#include "scene/gui/label.h"
```

**Review by akien-mga — APPROVED** [review]
Needs a rebase to solve a merge conflict, otherwise should be good to merge.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**